spec/graphql/language/document_from_schema_definition_spec.rb

```diff
@@ -0,0 +1,337 @@
+    enum __DirectiveLocation {
+      # Location adjacent to a query operation.
+      QUERY
+
+      # Location adjacent to a mutation operation.
+      MUTATION
+
+      # Location adjacent to a subscription operation.
+      SUBSCRIPTION
+
+      # Location adjacent to a field.
+      FIELD
+
+      # Location adjacent to a fragment definition.
+      FRAGMENT_DEFINITION
+
+      # Location adjacent to a fragment spread.
+      FRAGMENT_SPREAD
+
+      # Location adjacent to an inline fragment.
+      INLINE_FRAGMENT
+
+      # Location adjacent to a schema definition.
+      SCHEMA
+
+      # Location adjacent to a scalar definition.
+      SCALAR
+
+      # Location adjacent to an object type definition.
+      OBJECT
+
+      # Location adjacent to a field definition.
+      FIELD_DEFINITION
+
+      # Location adjacent to an argument definition.
+      ARGUMENT_DEFINITION
+
+      # Location adjacent to an interface definition.
+      INTERFACE
+
+      # Location adjacent to a union definition.
+      UNION
+
+      # Location adjacent to an enum definition.
+      ENUM
+
+      # Location adjacent to an enum value definition.
+      ENUM_VALUE
+
+      # Location adjacent to an input object type definition.
+      INPUT_OBJECT
+
+      # Location adjacent to an input object field definition.
+      INPUT_FIELD_DEFINITION
+    }
+
+    # Represents signed double-precision fractional values as specified by [IEEE
+    # 754](http://en.wikipedia.org/wiki/IEEE_floating_point).
+    scalar Float
+
+    # Represents a unique identifier that is Base64 obfuscated. It is often used to
+    # refetch an object or as key for a cache. The ID type appears in a JSON response
+    # as a String; however, it is not intended to be human-readable. When expected as
+    # an input type, any string (such as `"VXNlci0xMA=="`) or integer (such as `4`)
+    # input value will be accepted as an ID.
+    scalar ID
+
+    # Scalar description
+    scalar CustomScalar
+
+    enum Site {
+      DESKTOP
+      MOBILE
+    }
+
+    union Union = Type | QueryType
+
+    directive @skip(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
+
+    directive @include(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
+
+    # Marks an element of a GraphQL schema as no longer supported.
+    directive @deprecated(reason: String = "No longer supported") on FIELD_DEFINITION | ENUM_VALUE
+
+    schema {
+      query: QueryType
+      mutation: MutationType
+    }
+  GRAPHQL
+  }
+
+  let(:subject) { GraphQL::Language::DocumentFromSchemaDefinition }
+
+  describe "#document" do
+    it "returns the document AST from the given schema" do
+      schema = GraphQL::Schema.from_definition(schema_idl)
+      document = subject.new(schema).document
+
+      expected_document = GraphQL.parse(expected_complete_idl)
+
+      assert equivalent_node?(expected_document, document)
+    end
+  end
+
+  private
+
+  def equivalent_node?(expected, node)
```

Comment:
Just to help me understand, how is this different from [AbstractNode#eql?](https://github.com/rmosolgo/graphql-ruby/blob/7109e74dca7893016e326624dc1e1bef51b7d709/lib/graphql/language/nodes.rb#L35) ? 

Is it just that this one returns `true` even when children & scalars are in different _order_?

Reply:
Yep.. thats all 😓  but as discussed on slack,

Reply:
👌 thanks!